shared/src/main/scala/crystal/PotOption.scala

```diff
@@ -0,0 +1,111 @@
+package crystal
+
+import cats.syntax.all._
+import monocle.Prism
+
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+sealed trait PotOption[+A] {
+  def map[B](f: A => B): PotOption[B] =
+    this match {
+      case PotOption.Pending        => PotOption.Pending
+      case err @ PotOption.Error(_) => err.valueCast[B]
+      case PotOption.ReadyNone      => PotOption.ReadyNone
```

Comment:
When is it `ReadyNone`?

Reply:
In a topic, channel or queue stream, `ReadyNone` means the stream fiber has started (ie: messages received from now on will not be lost), but no messages have been received yet.

When communicating with a worker, this is the moment to make the request.